cpp/csp/python/adapters/websocketsadapterimpl.cpp

```diff
@@ -0,0 +1,86 @@
+#include <csp/adapters/websockets/ClientAdapterManager.h>
+#include <csp/engine/PushInputAdapter.h>
+#include <csp/python/Conversions.h>
+#include <csp/python/Exception.h>
+#include <csp/python/InitHelper.h>
+#include <csp/python/PyAdapterManagerWrapper.h>
+#include <csp/python/PyEngine.h>
+#include <csp/python/PyInputAdapterWrapper.h>
+#include <csp/python/PyOutputAdapterWrapper.h>
+
+using namespace csp::adapters::websockets;
+
+namespace csp::python
+{
+
+using namespace csp;
+using namespace csp::python;
+
+//AdapterManager
+csp::AdapterManager * create_websockets_adapter_manager( PyEngine * engine, const Dictionary & properties )
+{
+    return engine -> engine() -> createOwnedObject<ClientAdapterManager>( properties );
+}
+
+static InputAdapter * create_websockets_input_adapter( csp::AdapterManager * manager, PyEngine * pyengine, PyObject * pyType, PushMode pushMode, PyObject * args )
+{
+    auto & cspType = pyTypeAsCspType( pyType );
+
+    PyObject * pyProperties;
+    PyObject * type;
+    auto * websocketsManager = dynamic_cast<ClientAdapterManager*>( manager );
+    if( !websocketsManager )
+        CSP_THROW( TypeError, "Expected ClientAdapterManager" );
+
+    if( !PyArg_ParseTuple( args, "O!O!",
+                           &PyType_Type, &type,
+                           &PyDict_Type, &pyProperties ) )
+        CSP_THROW( PythonPassthrough, "" );
+
+    return websocketsManager -> getInputAdapter( cspType, pushMode, fromPython<Dictionary>( pyProperties ) );
+}
+
+static OutputAdapter * create_websockets_output_adapter( csp::AdapterManager * manager, PyEngine * pyengine, PyObject * args )
+{
+    auto * websocketsManager = dynamic_cast<ClientAdapterManager*>( manager );
+    if( !websocketsManager )
+        CSP_THROW( TypeError, "Expected ClientAdapterManager" );
```

Comment:
same comment, any reference to this adapter that would go up to the user should have websocket in the name